### PR TITLE
[SYCL] Add missing special values to exp(complex)

### PR DESCRIPTION
**Pull request by KornevNikita**
Follow-up to https://github.com/intel/llvm/pull/15672
This patch updates float behavior

### CODE REVIEW

File: libdevice/fallback-complex.cpp [hunk 1]
```diff
@@ -150,8 +150,14 @@ float __complex__ __devicelib_cexpf(float __complex__ z) {
         z_imag = NAN;
       return CMPLXF(z_real, z_imag);
     }
-  } else if (__spirv_IsNan(z_real) && (z_imag == 0.0f)) {
-    return z;
+  } else if (__spirv_IsNan(z_real)) {
+    if (z_imag == 0.0f)
+      return z;
+    else /* z_imag != 0.0f */
+      return CMPLX(NAN, NAN);
```
Comment (bader) [inline libdevice/fallback-complex.cpp:157]:
```suggestion
    /* z_imag != 0.0f */
    return CMPLX(NAN, NAN);
```
https://llvm.org/docs/CodingStandards.html#don-t-use-else-after-a-return

Comment (KornevNikita) [reply libdevice/fallback-complex.cpp]: Updated in [f898b54](https://github.com/intel/llvm/pull/15980/commits/f898b547b55d099ecedf1a47588a2efa139274df)

**Review by steffenlarsen — APPROVED** [review]

**Comment by jinge90** [comment]
Hi, @KornevNikita 
I just committed a little fix to remove lit failure on Linux but Windows failure is still there.
I will check Windows later. Things are different for Windows, we didn't provide any complex math functions for Windows since MSVC C++ headers included them, we just enabled these functions on Windows by providing some functions invoked in these complex functions.
Thanks very much.

**Comment by KornevNikita** [comment]
@jinge90 very appreciated! thanks!

**Comment by KornevNikita** [comment]
@intel/llvm-gatekeepers please merge

**Comment by sarnex** [comment]
can you improve the title?

**Comment by KornevNikita** [comment]
@sarnex is it OK?

**Pull request merged**